controlpanel/api/migrations/0010_DATA_add_tools.py

```diff
@@ -0,0 +1,80 @@
+# Generated by Django 2.2.4 on 2019-08-28 10:34
+
+import json
+import os
+import secrets
+
+from django.conf import settings
+from django.db import migrations
+
+
+TOOLS = [
+    {
+        'chart_name': 'rstudio',
+        'version': '2.1.1',
```

Comment:
One thing I'm thinking is that this schema would mean to add more than one "rstudio" entry for different versions (as it's not technically normalised). I suppose adding more tables would make things more complicated.

I like the values being version-specific in case we'll somehow to break compatibility.

The drawback of having this in the DB is that we'll have to add an entry to allow a release of a newer version of a chart. But I suppose this is a feature as we want to "whitelist" the versions we support (although as far as I can tell from quick look we still deploy latest version right?).

Alternative is we get this info "life" from the helm repo, but not sure it's a good idea as it would not be a curated list of tools we support anymore.

(I'm starting to have a look so this is just a comment with few random thoughts)

Reply:
I wasn't particularly intending to offer users multiple versions of a tool, but when we release a new version of a tool, existing deployments will be different and we should be able to show that difference.

I have changed the Helm command to pass `--version={version}`, so we will need to deal with this as soon as we have a new version of a tool chart.

I'm not keen to rely on the Helm repo or Tiller as a source of truth - Tiller has been removed in Helm 3 and I'm not even sure Helm is worth using. It would be better to add more metadata to the Kubernetes deployments/services/etc to identify the tool version - I think this is the direction Helm is going too.

Reply:
Makes sense, so we have one "good"/recommended version of a tool in the DB and when a user has an old/different version we offer them the option to "Upgrade" 👍